projects/packages/blaze/src/class-dashboard-rest-controller.php

```diff
@@ -556,6 +567,22 @@ public function get_dsp_site_campaigns( $req ) {
 		return $this->get_dsp_generic( sprintf( 'v1/sites/%d/campaigns', $site_id ), $req );
 	}
 
+	/**
+	 * Redirect GET requests to WordAds DSP Site Stats endpoint for the site.
+	 *
+	 * @param WP_REST_Request $req The request object.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function get_dsp_site_stats( $req ) {
+		$site_id = $this->get_site_id();
+		if ( is_wp_error( $site_id ) ) {
+			return array();
+		}
+
+		return $this->get_dsp_generic( sprintf( 'v1/sites/%d/stats', $site_id ), $req );
```

Comment:
I think it will be OK to take campaigns as an example here. In this function you are directly fetching `v1/sites/%d/stats` endpoints, but it should be 1.1, and the stats controller for v1.1 in the DSP doesn't have the `/sites` prefix. 

Like to the campaigns example using API version variable:
```
https://github.com/Automattic/jetpack/blob/f676e1cfddd309a841adafe1e2b958c21f0420da/projects/packages/blaze/src/class-dashboard-rest-controller.php#L534-L542
```

If you don't want to use API version as a variable, be sure to use the same version in this `get_dsp_generic` as the one you are registering in `register_rest_route`.